lib/web/resources.go

```diff
@@ -50,7 +50,7 @@ func (h *Handler) checkAccessToRegisteredResource(w http.ResponseWriter, r *http
 		return nil, trace.Wrap(err)
 	}
 
-	resourceKinds := []string{types.KindNode, types.KindDatabaseServer, types.KindAppServer, types.KindKubeServer, types.KindWindowsDesktop}
+	resourceKinds := []string{types.KindNode, types.KindDatabaseServer, types.KindAppServer, types.KindKubeServer, types.KindWindowsDesktop, types.KindSAMLIdPServiceProvider}
```

Comment:
Is this all that's needed? It works from my testing but I could be missing something.

Reply:
This works, but you might want to convert this to use `ListUnifiedResources` so that you can make one API call (instead of 6 in the worst case).

Reply:
Seems like that might do the trick.

Though, while we are at it, might I suggest couple things?

1. Can we change the name of this method? Because it says `checkAccessToRegisteredResource ` but inside the method we return `HasResource: true` even if user does not has access to resource.

```
res, err := clt.ListResources(...)
if err != nil {
	if trace.IsAccessDenied(err) {
		continue
	}
	return nil, trace.Wrap(err)
}

if len(res.Resources) > 0 {
	return checkAccessToRegisteredResourceResponse{
		HasResource: true,
	}, nil
}
```

Maybe `checkClusterResourceCount` or something similar?

2. Can we move the Web UI logic somewhere in the Unified resource view?

Currently, it is called inside the `TeleportContext` which is initialized very early in the UI bootstrap process. So even if there are resources, we will be needlessly firing the `resources/check` API the first time user signs in to Teleport (given enough privilege to add new resource as [checked here](https://github.com/gravitational/teleport/blob/8fb974f464bdc952da680cb4e0de17d0ea0328d6/web/packages/teleport/src/teleportContext.tsx#L101-L104)).

Reply:
I am not sure what changes would the 2nd suggestion entail, so given the scope of your PR, feel free to tackle it later if you wish to :)

Reply:
Gotcha, makes sense to me – the func name did confuse me at first. I'll look into using `ListUnifiedResources` here, & work on those other changes in another PR